test/src/test/java/jenkins/model/NodesTest.java

```diff
@@ -88,6 +90,28 @@ public void addNodeShouldReplaceExistingNode() throws Exception {
         assertThat(r.jenkins.getNode("foo"), sameInstance(newNode));
     }
 
+    @Test
+    @Issue("JENKINS-56403")
+    public void replaceNodeShouldRemoveOldNode() throws Exception {
```

Comment:
this test does fail after `git co 3df6ccaec8ba60b144d60a5302233e149507712b~ core/src/main/java`, so this is :+1: